react/features/app/middleware.js

```diff
@@ -63,36 +66,77 @@ function _connectionEstablished(store, next, action) {
 /**
  * Navigates to a route in accord with a specific redux state.
  *
- * @param {Object} state - The redux state which determines/identifies the route
+ * @param {Store} store - The redux store which determines/identifies the route
  * to navigate to.
  * @private
  * @returns {void}
  */
-function _navigate(state) {
+function _navigate({ dispatch, getState }) {
+    const state = getState();
     const { app, getRouteToRender } = state['features/app'];
     const routeToRender = getRouteToRender && getRouteToRender(state);
 
+    // Create/destroy the local tracks as needed: create them the first time we
+    // are going to render an actual route (be that the WelcomePage or the
+    // Conference).
+    //
+    // When the WelcomePage is disabled, the app will transition to the
+    // null/undefined route. Detect these transitions and create/destroy the
+    // local tracks so the camera doesn't stay open if the app is not rendering
+    // any component.
+    if (typeof routeToRender === 'undefined' || routeToRender === null) {
+        // Destroy the local tracks if there is no route to render and there is
+        // no welcome page.
+        app.props.welcomePageEnabled || dispatch(destroyLocalTracks());
+    } else {
+        // Create the local tracks if they haven't been created yet.
+        state['features/base/tracks'].some(t => t.local)
+            || dispatch(createInitialLocalTracks());
+    }
+
```

Comment:
The added source code above is specific to our React Native app and must not execute in our Web app.